data/src/main/java/com/example/data/repo/NewsRepositoryImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.example.data.repo;
+
+import androidx.annotation.NonNull;
+
+import com.example.data.R;
+import com.example.data.mapper.NewsDtoMapper;
+import com.example.data.mapper.NewsEntityToArticle;
+import com.example.data.model.Article;
+import com.example.data.repo.local.NewsLocalDataStore;
+import com.example.data.repo.remote.NewsRepository;
+import com.example.data.repo.remote.RemoteSource;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Scheduler;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class NewsRepositoryImpl implements NewsRepository {
+    RemoteSource source;
```

Comment:
we could pe making these fields private and final, also let's have same naming (remoteDataStore, localDataStore )